includes/admin/class-wc-rest-payments-reader-controller.php

```diff
@@ -62,4 +73,71 @@ public function get_summary( $request ) {
 		return rest_ensure_response( $summary );
 	}
 
+	/**
+	 * Proxies the get all readers request to the server.
+	 *
+	 * @param WP_REST_REQUEST $request Request object.
+	 * @return WP_REST_Response|WP_Error
+	 */
+	public function get_all_readers( $request ) {
+		try {
+			return rest_ensure_response( $this->fetch_readers() );
+		} catch ( API_Exception $e ) {
+			return rest_ensure_response( new WP_Error( $e->get_error_code(), $e->getMessage() ) );
+		}
+	}
+
+	/**
+	 * Check if the reader status is active
+	 *
+	 * @param array  $readers The readers charges object.
+	 * @param string $id      The reader ID.
+	 * @return bool
+	 */
+	private function is_reader_active( $readers, $id ) {
+		foreach ( $readers as $reader ) {
+			if ( $reader['reader_id'] === $id ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Attempts to read readers from transient and re-populates it if needed.
+	 *
+	 * @return array         Terminal readers.
+	 * @throws API_Exception If request to server fails.
+	 */
+	private function fetch_readers(): array {
+		$readers = get_transient( static::STORE_READERS_TRANSIENT_KEY );
+
+		if ( ! $readers ) {
+			// Retrieve terminal readers.
+			$readers_data = $this->api_client->get_terminal_readers();
+
+			// Retrieve the readers by charges.
+			$reader_by_charges = $this->api_client->get_readers_charge_summary( gmdate( 'Y-m-d', time() ) );
+
+			$readers = [];
+			foreach ( $readers_data as $reader ) {
+				$readers[] = [
+					'id'          => $reader['id'],
+					'livemode'    => $reader['livemode'],
+					'device_type' => $reader['device_type'],
+					'label'       => $reader['label'],
+					'location'    => $reader['location'],
+					'metadata'    => $reader['metadata'],
+					'status'      => $reader['status'],
+					'is_active'   => $this->is_reader_active( $reader_by_charges, $reader['id'] ),
+				];
+			}
+
+			set_transient( static::STORE_READERS_TRANSIENT_KEY, $readers, DAY_IN_SECONDS );
```

Comment:
Let's align with account caching and keep it for 2 hours only.

Reply:
Updated to 2 hours.